laravel4/doc/man/man3/User.3
.TH "User" 3 "Thu May 21 2015" "SINControl" \" -*- nroff -*-
.ad l
.nh
.SH NAME
User \- 
.SH SYNOPSIS
.br
.PP
.PP
Inherits Eloquent, UserInterface, and RemindableInterface\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBsetPassword\fP ($password)"
.br
.ti -1c
.RI "\fBgetUserInfo\fP ()"
.br
.ti -1c
.RI "\fBuser_info\fP ()"
.br
.ti -1c
.RI "\fBgetGroup\fP ()"
.br
.ti -1c
.RI "\fBgetGroups\fP ()"
.br
.ti -1c
.RI "\fBisGroupMember\fP ($group)"
.br
.ti -1c
.RI "\fBgetLowestGid\fP ()"
.br
.ti -1c
.RI "\fBhasExpired\fP ()"
.br
.ti -1c
.RI "\fBgetAuthIdentifier\fP ()"
.br
.ti -1c
.RI "\fBgetAuthPassword\fP ()"
.br
.ti -1c
.RI "\fBgetRememberToken\fP ()"
.br
.ti -1c
.RI "\fBsetRememberToken\fP ($value)"
.br
.ti -1c
.RI "\fBgetRememberTokenName\fP ()"
.br
.ti -1c
.RI "\fBgetReminderEmail\fP ()"
.br
.in -1c
.SS "Data Fields"

.in +1c
.ti -1c
.RI "\fB$timestamps\fP = false"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "\fB$table\fP = 'user'"
.br
.ti -1c
.RI "\fB$hidden\fP = array ('crypt')"
.br
.in -1c
.SH "Member Function Documentation"
.PP 
.SS "getAuthIdentifier ()"
Get the unique identifier for the user\&.
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.SS "getAuthPassword ()"
Get the password for the user\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.SS "getGroup ()"

.SS "getGroups ()"

.SS "getLowestGid ()"

.SS "getRememberToken ()"
Get the token value for the 'remember me' session\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.SS "getRememberTokenName ()"
Get the column name for the 'remember me' token\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.SS "getReminderEmail ()"
Get the e-mail address where password reminders are sent\&.
.PP
\fBReturns:\fP
.RS 4
string 
.RE
.PP

.SS "getUserInfo ()"

.SS "hasExpired ()"

.SS "isGroupMember ( $group)"

.SS "setPassword ( $password)"

.SS "setRememberToken ( $value)"
Set the token value for the 'remember me' session\&.
.PP
\fBParameters:\fP
.RS 4
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.SS "user_info ()"

.SH "Field Documentation"
.PP 
.SS "$hidden = array ('crypt')\fC [protected]\fP"

.SS "$table = 'user'\fC [protected]\fP"

.SS "$timestamps = false"


.SH "Author"
.PP 
Generated automatically by Doxygen for SINControl from the source code\&.


laravel4/app/models/User.php
<?php

use Illuminate\Auth\UserInterface;
use Illuminate\Auth\Reminders\RemindableInterface;

class User extends Eloquent implements UserInterface, RemindableInterface
{

	/**
	 * The database table used by the model.
	 *
	 * @var string
	 */
	protected $table = 'user';
	public $timestamps = false;

	/**
	 * The attributes excluded from the model's JSON form.
	 *
	 * @var array
	 */
	protected $hidden = array ('crypt', 'smb_lm', 'smb_nt', 'remember_token');
	
	public function setPassword ($password)
	{
		$this->crypt = crypt ($password, '$6$rounds=' . mt_rand (8000, 12000) . '$' . bin2hex (openssl_random_pseudo_bytes (64)) . '$');
	}
	
	public function userInfo ()
	{
		return $this->belongsTo ('UserInfo');
	}
	
	public function mailDomainVirtual ()
	{
		return $this->hasMany ('MailDomainVirtual');
	}
	
	public function mailForwardingVirtual ()
	{
		return $this->hasMany ('MailForwardingVirtual');
	}
	
	public function mailUserVirtual ()
	{
		return $this->hasMany ('MailUserVirtual');
	}
	
	public function getGroup ()
	{
		return Group::where ('gid', $this->gid)->first ();
	}
	
	public function getGroups ()
	{
		return UserGroup::where ('uid', $this->uid)->get ();
	}
	
	public function primaryGroup ()
	{
		return $this->hasOne ('Group', 'gid', 'gid');
	}
	
	public function isGroupMember ($group)
	{
		return (UserGroup::where ('uid', $this->uid)->where ('gid', $group->gid)->count () > 0);
	}
	
	public function getLowestGid () // Lagere gid betekent hogere permissies //
	{
		$userGroups = UserGroup::where ('uid', $this->uid);
		$lowestGid = $this->gid;
		
		if ($userGroups->count () > 0 && $userGroups->min ('gid') < $lowestGid)
			$lowestGid = $userGroups->min ('gid');
		
		return $lowestGid;
	}
	
	public function hasExpired ()
	{
		$now = ceil (time () / 60 / 60 / 24);
		
		return ($this->expire <= $now && $this->expire != -1);
	}
	
	public function vhost ()
	{
		return $this->hasMany ('ApacheVhostVirtual', 'uid', 'uid');
	}
	
	public function url ()
	{
		return action ('StaffUserController@more', $this->id);
	}
	
	public function link ()
	{
		return '<a href="' . $this->url () . '">' . get_class () . '#' . $this->id . ($this->userInfo != NULL ? ' (' . $this->userInfo->username . ')' : '') . '</a>';
	}
	
	public function calculateDiskUsage ($save = false)
	{
		$exitStatus = NULL;
		$cmd = 'du -shbxB 1048576 ' . escapeshellarg ($this->homedir);
		
		$output = array ();
		exec ($cmd, $output, $exitStatus);

		if ($exitStatus == 0)
		{
			$usage = explode ("\t", implode (PHP_EOL, $output))[0];
			
			if ($save)
			{
				$this->diskusage = $usage;
				
				$this->save ();
			}
			
			return $usage;
		}
	}
	
	public static function calculateAndSaveDiskUsage ()
	{
		$now = time () / 60 / 60 / 24;
		
		foreach (User::where ('expire', '>', $now)->orWhere ('expire', -1)->get () as $user)
			$user->calculateDiskUsage (true);
	}

	/**
	 * Get the unique identifier for the user.
	 *
	 * @return mixed
	 */
	public function getAuthIdentifier ()
	{
		return $this->getKey ();
	}

	/**
	 * Get the password for the user.
	 *
	 * @return string
	 */
	public function getAuthPassword ()
	{
		return $this->password;
	}

	/**
	 * Get the token value for the "remember me" session.
	 *
	 * @return string
	 */
	public function getRememberToken ()
	{
		return $this->remember_token;
	}

	/**
	 * Set the token value for the "remember me" session.
	 *
	 * @param  string  $value
	 * @return void
	 */
	public function setRememberToken ($value)
	{
		$this->remember_token = $value;
	}

	/**
	 * Get the column name for the "remember me" token.
	 *
	 * @return string
	 */
	public function getRememberTokenName ()
	{
		return 'remember_token';
	}

	/**
	 * Get the e-mail address where password reminders are sent.
	 *
	 * @return string
	 */
	public function getReminderEmail ()
	{
		return $this->email;
	}
	
	public function __toString ()
	{
		$name = '#' . $this->id;
		
		if ($this->userInfo != NULL)
			$name = $this->userInfo->username;
		
		return 'Gebruiker: ' . $name;
	}

}
